src/resources/js/views/events-bar.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * Makes sure we have all the required levels on the Tribe Object
+ *
+ * @since TBD
+ *
+ * @type   {PlainObject}
+ */
+tribe.events = tribe.events || {};
+tribe.events.views = tribe.events.views || {};
+
+/**
+ * Configures Events Bar Object in the Global Tribe variable
+ *
+ * @since TBD
+ *
+ * @type   {PlainObject}
+ */
+tribe.events.views.eventsBar = {};
+
+/**
+ * Initializes in a Strict env the code that manages the Event Views
+ *
+ * @since TBD
+ *
+ * @param  {PlainObject} $   jQuery
+ * @param  {PlainObject} obj tribe.events.views.manager
+ *
+ * @return {void}
+ */
+( function( $, obj ) {
+	'use strict';
+	var $document = $( document );
+
+	/**
+	 * Selectors used for configuration and setup
+	 *
+	 * @since TBD
+	 *
+	 * @type {PlainObject}
+	 */
+	obj.selectors = {
+		inputKeyword: '.tribe-common-c-search__input--keyword',
+		inputKeywordFocus: '.tribe-common-c-search__input--keyword-focus',
+		inputLocation: '.tribe-common-c-search__input--location',
+		inputLocationFocus: '.tribe-common-c-search__input--location-focus',
+	};
+
+	/**
+	 * Toggles input class if it has a value
+	 *
+	 * @since TBD
+	 *
+	 * @param {Event} event event object of click event
+	 *
+	 * @return {void}
+	 */
+	obj.setInputFocusClass = function( event ) {
+		var $input = event.data.target;
+
+		// Set the focus class if it has content.
+		$input.toggleClass( event.data.inputClassFocus, '' !== $input.val().trim() );
+	};
+
+	/**
+	 * Binds events for the events bar change listeners
+	 *
+	 * @since TBD
+	 *
+	 * @param {Event} event event object for 'afterSetup.tribeEvents' event
+	 * @param {integer} index jQuery.each index param from 'afterSetup.tribeEvents' event
+	 * @param {jQuery} $container jQuery object of view container
+	 * @param {object} data data object passed from 'afterSetup.tribeEvents' event
+	 *
+	 * @return {void}
+	 */
+	obj.bindEvents = function( event, index, $container, data ) {
+
+		// Bind event for the keyword input.
+		$container
+			.find( obj.selectors.inputKeyword )
+			.each( function( index, input ) {
+				$( input ).toggleClass( obj.selectors.inputKeywordFocus.className(), '' !== $( input ).val().trim() );
+				$( input ).on( 'change', { target: $( this ), inputClassFocus: obj.selectors.inputKeywordFocus.className() }, obj.setInputFocusClass );
+			} );
+
+		// Bind event for the location input.
+		$container
+			.find( obj.selectors.inputLocation )
+			.each( function( index, input ) {
```

Comment:
Each one of these binds should live into a separate method.

Reply:
@bordoni was this resolved? did you mean that the `each()` should be calling something like `obj.bindInputChangeListener` rather than an anonymous `function()`?